src/Microsoft.TestPlatform.CrossPlatEngine/DataCollection/InProcDataCollectionExtensionManager.cs

```diff
@@ -61,7 +61,7 @@ public InProcDataCollectionExtensionManager(string runSettings, ITestRunCache te
             // Initialize InProcDataCollectors
             this.InitializeInProcDataCollectors(runSettings);
 
-            FlushResultTestResultPoperty = TestProperty.Register(id: "allowTestResultFlush", label: "allowTestResultFlush", category: string.Empty, description: string.Empty, valueType: typeof(bool), validateValueCallback: null, attributes: TestPropertyAttributes.None, owner: typeof(TestCase));
+            FlushResultTestResultPoperty = TestProperty.Register(id: "allowTestResultFlush", label: "allowTestResultFlush", category: string.Empty, description: string.Empty, valueType: typeof(bool?), validateValueCallback: null, attributes: TestPropertyAttributes.None, owner: typeof(TestCase));
```

Comment:
Value of FlushResultTestResultPoperty is null when inproc data collection is not enabled.
Otherwise, we might always get false by default, which will disable the flushing of test results permanently, vstest.console process will never get the test results even though tests are executed as expected.